exporter/prometheusremotewriteexporter/exporter.go

```diff
@@ -274,11 +295,25 @@ func (prwe *PrwExporter) export(ctx context.Context, tsMap map[string]*prompb.Ti
 		return errs
 	}
 
-	input := make(chan *prompb.WriteRequest, len(requests))
-	for _, request := range requests {
-		input <- request
+	if !prwe.walEnabled() {
+		return prwe.exportWriteRequests(ctx, requests)
 	}
-	close(input)
+
+	// Otherwise, serialize the requests to the WAL.
+	if err := prwe.writeToWAL(requests); err != nil {
+		errs = append(errs, consumererror.Permanent(err))
+	}
+	return errs
+}
+
+func (prwe *PrwExporter) exportWriteRequests(ctx context.Context, requests []*prompb.WriteRequest) (errs []error) {
+	input := make(chan *prompb.WriteRequest, len(requests))
+	go func() {
+		defer close(input)
+		for _, request := range requests {
+			input <- request
+		}
+	}()
```

Comment:
Why async? What is the motivation for this change?

Reply:
We want to begin processing requests immediately without having to wait to fill up the buffered channel. However, I shall revert it.